### DISCUSSION TITLE
Publishing other types of files (other than markdown files)

### DISCUSSION THREAD

**Discussion by Neutronicsenthusiast064** [alangrainger/share-note#38] [Vote for upcoming features]
Hello, following my message in Issue #34, I have a suggestion to submit.

As a student, I have a lot of books and documents in pdf/epub formats, and I often need to share these documents with other students or even with my professors (yes, I could continue spending hours uploading documents to my Google Drive to share them, but this plugin has an huge potential, so why not trying to enhance it?). 

It would also be great, **if it's possible**, to display MATLAB or Python files to share codes (these are just examples, as it could be interesting for everyone to share codes in C, C++, etc...).

By combining this idea and idea #27, we can create an online shareable Obsidian vault supporting different file types.

**Comment by alangrainger** [comment]
This plugin is certainly the wrong solution for the problem of sharing *files* with people. This is for sharing your notes with people.

In terms of "spending hours uploading" - if you were to use Share Note, you'd need to share each document individually one-by-one. With Google Drive or any other file-sharing software, you can drag 1,000 files into a shared folder with a single click. So I think the existing solutions for file sharing are going to be a much better bet for this application.
